routes/cityListings.js

```diff
@@ -851,6 +855,89 @@ router.post(
     }
 );
 
+router.post(
+    "/:id/vote",
+    async function (req, res, next) {
+        const listingId = req.params.id;
+        const cityId = req.cityId;
+        if (!cityId) {
+            return next(new AppError(`City is not present`, 404));
+        } else {
+            try {
+                const response = await database.get(tables.CITIES_TABLE, {
+                    id: cityId,
+                });
+                if (response.rows && response.rows.length === 0) {
+                    return next(new AppError(`City '${cityId}' not found`, 404));
+                }
+            } catch (err) {
+                return next(new AppError(err));
+            }
+        }
+
+        if (isNaN(Number(listingId)) || Number(listingId) <= 0) {
+            next(new AppError(`Invalid ListingsId ${listingId} given`, 400));
+            return;
+        }
+
+        const optionIdStr = req.body.optionId;
+        if (!optionIdStr) {
+            return next(new AppError(`OptionId not provided`, 400));
+        }
+        const optionId = Number(optionIdStr);
+        if (isNaN(optionId) || optionId <= 0) {
+            return next(new AppError(`Invalid OptionId ${optionId} given`, 400));
+        }
+
+        const requestVote = req.body.vote;
+        if (!requestVote) {
+            return next(new AppError(`Vote not provided`, 400));
+        }
+        const vote = requestVote === "up" ? 1 : -1;
```

Comment:
in this wouldn't it be easier to just get the int as vote instead of "up"
like in request {
vote: 1 // for up vote
vote: 0 // for down vote
}

Reply:
Using a string makes more sense because it makes it easier for the developer who will read the API and implement it in the app. And, even if we use an integer, we still have to check it like `const vote = requestVote ? 1 : -1`.

Reply:
For developers, I think 1 and 0 make more sense, and as far as this `const vote = requestVote? 1: -1` is concerned this is fine.